### ISSUE TITLE
Should we have props be set to an attribute called props?

### ISSUE THREAD

**Issue by BigAB** [BigAB/react-view-models#4]
It may be a good idea to do `vm.set({ props: props })` rather than calling `vm.set(props)` when the component receives props.  Maybe.

Though it seemed like a good idea to take the props passed into a **Connected Container Component** and just mix them into the viewModel with `vm.set(props)`, it actually ends up forcing the user to write weird code like this:

``` javascript
const ViewModel = DefineMap.extend({
  didHaveItsCallbackCalled: {
    type: 'boolean'
  },
  callback: {
    type: 'any',
    set(newVal) {
      return (ev) => {
        this.didHaveItsCallbackCalled = true;
        return newVal(ev);
      }
    }
  }
});
```

but would this be better?:

```
const ViewModel = DefineMap.extend({
  didHaveItsCallbackCalled: {
    type: 'boolean'
  },
  callback(ev)
    this.didHaveItsCallbackCalled = true;
    return this.props.callback(ev);
  }
});
```

Another more serious problem, is that, in the current implementation, props always "win".  
You could have the viewModels values change, but because everytime the new props are computed the parents props get set on viewModel, the view model always wins. See this example:  
https://gist.github.com/BigAB/b94f58a634c64453e08ab61c26f764c2

Solving that problem by only setting the viewModel values when "props are received", might end up being an even more confusing situation, as you get into this strange situation where outer forces are changing the VM value and then suddenly props get updated and wipe out other changes. It becomes very strange. but then again, that's how react works now, so maybe that's a non-issue.

Also, the current way, all props get passed to the child **Presentational Component** by default and it becomes annoying to bypass that for individual props, but this change makes it so no props are passed by default and you must explicitly write a serialized getter to pass props through:

```
const ViewModel = DefineMap.extend({
  bar: {
    get() {
      return this.props.bar;
    },
    serialize: true
  }
});
```

So... yeah, not sure which way to go on this one.

**Comment by BigAB** [comment]
Actually, I am not sure this is a good idea.

**Comment by BigAB** [comment]
Closed in favour of #6
